File: apps/server/aetherblog-service/blog-service/src/main/java/com/aetherblog/blog/entity/Post.java
package com.aetherblog.blog.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

/**
 * 文章实体
 * 
 * @ref §4.2 - 核心表结构 (V2 增强版)
 */
@Data
@Entity
@Table(name = "posts", indexes = {
    @Index(name = "idx_posts_slug", columnList = "slug"),
    @Index(name = "idx_posts_status", columnList = "status"),
    @Index(name = "idx_posts_published_at", columnList = "published_at"),
    @Index(name = "idx_posts_category", columnList = "category_id"),
    @Index(name = "idx_posts_author", columnList = "author_id"),
    @Index(name = "idx_posts_deleted", columnList = "deleted"),
    @Index(name = "idx_posts_embedding_status", columnList = "embedding_status")
})
public class Post {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 200)
    private String title;

    @Column(nullable = false, unique = true, length = 200)
    private String slug;

    /**
     * Markdown 原始内容
     */
    @Column(name = "content_markdown", columnDefinition = "TEXT")
    private String contentMarkdown;

    /**
     * 渲染后的 HTML 内容缓存
     */
    @Column(name = "content_html", columnDefinition = "TEXT")
    private String contentHtml;

    @Column(length = 500)
    private String summary;

    @Column(name = "cover_image", length = 500)
    private String coverImage;

    @Column(nullable = false, length = 20)
    @Enumerated(EnumType.STRING)
    private PostStatus status = PostStatus.DRAFT;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private Category category;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "author_id")
    private User author;

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(
        name = "post_tags",
        joinColumns = @JoinColumn(name = "post_id"),
        inverseJoinColumns = @JoinColumn(name = "tag_id")
    )
    private Set<Tag> tags = new HashSet<>();

    @Column(name = "view_count", nullable = false)
    private Long viewCount = 0L;

    @Column(name = "comment_count", nullable = false)
    private Long commentCount = 0L;

    @Column(name = "like_count", nullable = false)
    private Long likeCount = 0L;

    /**
     * 文章字数
     */
    @Column(name = "word_count")
    private Integer wordCount = 0;

    /**
     * 预计阅读时间 (分钟)
     */
    @Column(name = "reading_time")
    private Integer readingTime = 0;

    /**
     * 是否置顶
     */
    @Column(name = "is_pinned", nullable = false)
    private Boolean isPinned = false;

    /**
     * 置顶优先级 (数值越大优先级越高，0表示不置顶)
     */
    @Column(name = "pin_priority", nullable = false)
    private Integer pinPriority = 0;

    /**
     * 是否精选
     */
    @Column(name = "is_featured", nullable = false)
    private Boolean isFeatured = false;

    /**
     * 是否允许评论
     */
    @Column(name = "allow_comment", nullable = false)
    private Boolean allowComment = true;

    /**
     * 文章密码 (加密访问)
     */
    @Column(length = 100)
    private String password;

    // ========== V2 新增字段 ==========

    /**
     * SEO 标题
     */
    @Column(name = "seo_title", length = 200)
    private String seoTitle;

    /**
     * SEO 描述
     */
    @Column(name = "seo_description", length = 300)
    private String seoDescription;

    /**
     * SEO 关键词
     */
    @Column(name = "seo_keywords", length = 200)
    private String seoKeywords;

    /**
     * 向量化状态
     */
    @Column(name = "embedding_status", nullable = false, length = 20)
    @Enumerated(EnumType.STRING)
    private EmbeddingStatus embeddingStatus = EmbeddingStatus.PENDING;

    /**
     * 软删除标记
     */
    @Column(nullable = false)
    private Boolean deleted = false;

    /**
     * 定时发布时间
     */
    @Column(name = "scheduled_at")
    private LocalDateTime scheduledAt;

    @Column(name = "published_at")
    private LocalDateTime publishedAt;

    @CreationTimestamp
    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    public enum PostStatus {
        DRAFT, PUBLISHED, ARCHIVED, SCHEDULED
    }

    public enum EmbeddingStatus {
        PENDING, INDEXED, FAILED
    }
}

File: apps/server-go/internal/model/post.go
// Package model 定义与 PostgreSQL 数据库表一一对应的数据模型结构体。
// 字段通过 sqlx 的 `db` 标签与数据库列名对应，可空列使用指针类型表示。
package model

import "time"

// Post 对应数据库 `posts` 表，是博客系统的核心内容单元。
type Post struct {
	ID                 int64      `db:"id"`
	Title              string     `db:"title"`
	Slug               string     `db:"slug"`                 // URL 友好的唯一标识符，不可为空
	ContentMarkdown    *string    `db:"content_markdown"`     // Markdown 原始源文，导入的外部文章可能为 nil
	ContentHTML        *string    `db:"content_html"`         // 预渲染的 HTML 内容；尚未渲染时为 nil
	Summary            *string    `db:"summary"`              // 文章摘要；nil 时由系统在展示时自动生成
	CoverImage         *string    `db:"cover_image"`          // 封面图片 URL；无封面时为 nil
	Status             string     `db:"status"`               // 文章状态：DRAFT（草稿）| PUBLISHED（已发布）| ARCHIVED（已归档）| SCHEDULED（定时发布）
	CategoryID         *int64     `db:"category_id"`          // 所属分类 ID（外键）；nil 表示未分类
	AuthorID           *int64     `db:"author_id"`            // 作者用户 ID（外键）；历史导入文章可能为 nil
	ViewCount          int64      `db:"view_count"`           // 文章浏览次数
	CommentCount       int64      `db:"comment_count"`        // 文章评论数
	LikeCount          int64      `db:"like_count"`           // 文章点赞数
	WordCount          int        `db:"word_count"`           // Markdown 内容的字符数（保存时计算）
	ReadingTime        int        `db:"reading_time"`         // 预估阅读时间（分钟），按 300 字/分钟计算
	IsPinned           bool       `db:"is_pinned"`            // 是否置顶；置顶文章在列表中优先显示
	PinPriority        int        `db:"pin_priority"`         // 置顶排序权重；值越大越靠前
	IsFeatured         bool       `db:"is_featured"`          // 是否为精选文章
	IsHidden           bool       `db:"is_hidden"`            // 是否隐藏；隐藏文章不出现在公开列表中
	AllowComment       bool       `db:"allow_comment"`        // 是否允许评论
	Password           *string    `db:"password"`             // 访问密码的 bcrypt 哈希值；nil 表示公开文章
	SEOTitle           *string    `db:"seo_title"`            // SEO 自定义标题，可为空
	SEODescription     *string    `db:"seo_description"`      // SEO 元描述，可为空
	SEOKeywords        *string    `db:"seo_keywords"`         // SEO 关键词，可为空
	EmbeddingStatus    string     `db:"embedding_status"`     // 向量嵌入状态：PENDING（待处理）| INDEXED（已索引）| FAILED（失败）
	Deleted            bool       `db:"deleted"`              // 软删除标志；已删除文章在所有查询中被排除
	ScheduledAt        *time.Time `db:"scheduled_at"`         // 定时发布的未来时间点；仅 SCHEDULED 状态文章有效
	PublishedAt        *time.Time `db:"published_at"`         // 实际发布时间戳；未发布文章为 nil
	SourceKey          *string    `db:"source_key"`           // 可选的外部来源标识符（如导入任务 ID）
	LegacyAuthorName   *string    `db:"legacy_author_name"`   // 从旧系统迁移时保留的作者名称
	LegacyVisitedCount int64      `db:"legacy_visited_count"` // 从旧系统迁移时保留的历史访问计数
	LegacyCopyright    *string    `db:"legacy_copyright"`     // 从旧系统迁移时保留的版权文本
	CreatedAt          time.Time  `db:"created_at"`           // 记录创建时间
	UpdatedAt          time.Time  `db:"updated_at"`           // 记录最后更新时间
}
